UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/NMDataService.cs

```diff
@@ -159,5 +159,38 @@ public async Task<ShowWeeklyFilesResponseModel> GetWeeklyFilesResponseModelsAsyn
 
             return showWeeklyFilesResponses;
         }
+        public async Task<byte[]> DownloadFssFileAsync(string batchId, string fileName, string correlationId)
+        {
+            try
+            {
+                _logger.LogInformation(EventIds.GetSingleWeeklyNMFileStarted.ToEventId(), "Maritime safety information request to get single weekly NM file started for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", batchId, fileName, correlationId);
+
+                string accessToken = await _authFssTokenProvider.GenerateADAccessToken(correlationId);
+
+                Stream stream = await _fileShareService.FSSDownloadFileAsync(batchId, fileName, accessToken, correlationId);
+
+                byte[] fileBytes = new byte[stream.Length + 10];
+
+                int numBytesToRead = (int)stream.Length;
+                int numBytesRead = 0;
+                do
+                {
+                    int n = stream.Read(fileBytes, numBytesRead, 10);
```

Comment:
Let's stick with the async read method and ask for a all bytes rather than just 10 at a time please:
```suggestion
                    int n = await stream.ReadAsync(fileBytes, numBytesRead, numBytesToRead);
```
Last change I promise 😃

Reply:
Changes done as per required.